src/ServiceControl/Operations/ErrorProcessor.cs

```diff
@@ -117,7 +117,7 @@ async Task ProcessMessage(MessageContext context, IIngestionUnitOfWork unitOfWor
 
                 var processingAttempt = failedMessageFactory.CreateProcessingAttempt(
                     context.Headers,
-                    new Dictionary<string, object>(metadata), // TODO: metadata is already a dictionary, it this really needed?
+                    new Dictionary<string, object>(metadata),
```

Comment:
I think this is attempting to duplicate the dictionary so modifications don't affect the original on any sort of retry.